core/vm/interface.go

```diff
@@ -74,6 +74,8 @@ type StateDB interface {
 	AddPreimage(common.Hash, []byte)
 
 	ForEachStorage(common.Address, func(common.Hash, common.Hash) bool) error
+
+	Finalise(bool)
```

Comment:
Why is this needed? Changes in this commit only use `Finalise` from statedb right?

Reply:
We need to use Finalise method of the current vmenv state which is possible only if this is enabled. You can try removing this in code and will get the reason why this was added. We can discuss this offline if needed ✅